### DISCUSSION TITLE
Availability of Etsy search and trending data

### DISCUSSION THREAD

**Discussion by itiievskyi** [etsy/open-api#930] [Q&A]
Hi, 
I develop a service that works with Etsy through the API using client credentials. Etsy API v3 has limited functionality but it's enough for most tasks. The current problem is that I can't get the full information that customer could be interested in: stats of internal Etsy searches with the same tags as the listing has, most searched words/tags, etc. I would suppose there is no such possibility but a lot of services provide that info. For example, eRank. And it's not endorsed or certified by Etsy. So, what is the source of such data if it's unavailable through official API?

It looks like trending products are going to become unavailable in a week, when v2 is unsupported.

Also, I found some repos that use `https://www.etsy.com/suggestions_ajax.php?search_query=` that responses with the list of suggestions for user when they use an Etsy search. Is it safe to use it? It doesn't require any auth but I suppose it may check headers for other parameters and there is a possibility to get banned by ip due to frequent requests.

Or maybe there are some other API resources that can offer such sort of data?

**Comment by showtroylove** [comment]
We would like to have access to such data too.   This information is crucial to the product suite we're developing.  Etsy appears to be holding this information close to the vest, if you will.  Please share whatever information you find regarding official channels to obtain this information.  Much appreciated.

**Comment by avisrael** [comment]
I think its the golden data. Everybody wants it but etsy doesn't share. I found another thread about it from beginning 2022. 
Erank, use only "results" as competition data. the data, how many people search it..i think its just estimation. it is fake just as everbee admitted the sales number, is just an estimation (its really bad in my opinion, very bad estimation, i checked it) but everybody rely on these numbers. Erank might take the number from google maybe ? 
as for Tags. I wouldn't trust tags that are created by sellers. They don't rely on actual data that shows anybody actually search these tags. Many tags are not even autocompleted in the search bar. it just seller created data. 
Also Etsy doesn't rely on these tags when they advertise your listing. They take word combinations from title and tags and assign it to "searches/keywords" that might be related.

**Comment by showtroylove** [comment]
I also got the impression Etsy didn't share this data, at least, officially.  I've wondered about the accuracy of tools like EverBee.  I believe it is possible to calculate quality estimates using data from the Etsy API and other sources, however.

**Reply by avisrael** [reply]
my guess. once you have stats of real shop views/favorites/sales  you can create algorithm to predict sales based on view/favorites. They also get from users access to the user's shop data, so they can fine tune this algo..but i found its really not accurate.

**Comment by itiievskyi** [comment]
It's interesting that on different services, some data that seems inaccessible coincides. For example, a long-tail tag tool shows approximately the same set of tags for a given query. But it's a way longer than what can be obtained from the `https://www.etsy.com/suggestions_ajax.php?search_query=`. Moreover, there are services that assign a score to these tags (where could it come from?).

**Reply by avisrael** [reply]
exactly. both everbee and erank give you search estimations on suggested keyword. I am glad to find i am not the only one asking the same questions

**Comment by PMStanley** [comment]
eRank have answered this a few times, but they always close the discussion down. If I remember correctly they buy information from an agency who have a panel of shoppers who provide information about what they buy back to the agency and sell that data to eRank (and others). 

It's useful trend data for large markets but I question its validity for making specific decisions for individual shops and items. eRank haven't shared any information on the panel data they use, which means there's no indication of how large the panel is or any other information which would be key for sellers to make educated decisions on (size alone is not enough to make good decisions at a single shop scale).  

I would urge caution in this space, there's a few outlets out there offering recommendations without fully explaining their data sources and are potentially making bad recommendations to their customers without the customers realising it's bad.

**Reply by avisrael** [reply]
its about the money. we all know..
the strange part is etsy. Why would they hide the numbers ? if people would know the real numbers all the niches with little competition will be filled. more sellers will be happy. Google give away this info and this created all the SEO industry, it just made google stronger because people knew where to invest money in ads.

**Comment by itiievskyi** [comment]
Etsy has a complicated ranking algorithm, I believe, and the real importance of tags, title, shop rating, listing age, etc. is still unclear.
Moreover, the search results depend on user as well. Even trying search by the same keywords from the same IP address with and without authorization I get completely different results. 
By the way, V3 endpoint `findAllListingsActive` has a sorting on `score` whatever it is.

**Reply by avisrael** [reply]
i myself ranked one 1st page of search results, it had low competition, it even had a bestseller on that page, very good sales. even though i ranked high, i hardly get views. it means the keywords are not searched enough. i bombed title/tags/desc with the same keywords. Even description is important because before adding the keywords to desc, i didn't rank. 
Etsy save browse history so the default search results are modified. if you search from incognito, you will get more consistent results. I got this tip from some youtube guru, the research in incognito so they will not effect search results. Also if you search without a username, you get results with no ads. if you login..you get ads. 
But still the info about the keyword, volume of search is the most critical here. And we don't have it
